src/signalrclient/connection_impl.cpp

```diff
@@ -418,7 +418,10 @@ namespace signalr
                 }
             });
 
-        timer(m_scheduler, [connect_request_done, connect_request_lock, callback](std::chrono::milliseconds duration)
+        auto negotiate_timeout = std::chrono::seconds(5);
```

Comment:
This is pretty aggressive compared to the JS and .NET client, right? Should we make this bigger and/or configurable?

Reply:
Yeah, I can remove this change, it's unrelated, trying to do smaller changes, and this one is focused on handshake.

Reply:
Don't revert it. I think it reads more clearly this way and you didn't change the value. I just noticed it for the first time due to this PR. I get not wanting to slow down the PR based on this. We can file a follow up issue or we can wait for a customer to ask for it.

Reply:
If you want to up it to 120 seconds to match HttpConnection.HttpClientTimeout, I'm also perfectly find with that, but no need to debate it here if you disagree.

Reply:
> We can file a follow up issue or we can wait for a customer to ask for it.

I'll mention it on the issue https://github.com/dotnet/aspnetcore/issues/30495

Reply:
> If you want to up it to 120 seconds to match HttpConnection.HttpClientTimeout

Actually looking at this more, this 5 second timeout was part of porting the old client. It doesn't exist in the other clients, and I've heard no one asking for it. I'm working on another change right now centered around negotiate, so I'll fix it up there.

Reply:
Let's just undo all the changes in this file before merging since it's fixed in another PR.